### PR TITLE
Add support for templates on dropdowns

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala [hunk 1]
```diff
@@ -65,6 +72,20 @@ object DropdownBase {
       .applyOrNot(props.disabled, _.disabled(_))
       .applyOrNot(props.dropdownIcon, _.dropdownIcon(_))
       .applyOrNot(props.tooltip, _.tooltip(_))
+      .applyOrNot(
+        props.itemTemplate,
+        (c, p) =>
+          c.itemTemplate(raw =>
+            p(props.selectItemFinder(raw.asInstanceOf[CSelectItem].value)).rawNode
```
Comment (rpiaggio) [inline prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala:79]: I'm guessing this is to deal with unfriendly types from ST.

**Review by rpiaggio — APPROVED** [review]
LGTM

**Review by toddburnside — APPROVED** [review]
LGTM

File: prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala [hunk 2]
```diff
@@ -65,6 +72,20 @@ object DropdownBase {
...
+          )
+      )
+      .applyOrNot(
+        props.valueTemplate,
```
Comment (toddburnside) [inline prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala:83]: Have you tested this for `DropdownOptional`? Seems like `selectItemFinder` might fail?

File: prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala (hunk 1 above)
Comment (cquiroz) [reply prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala:79]: Yes

File: prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala (hunk 2 above)
Comment (cquiroz) [reply prime-react/src/main/scala/lucuma/react/primereact/DropdownBase.scala:83]: Good catch, I added an extra param to support handling empty cases

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**